src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -148,12 +148,12 @@ protected override bool TrySerializeKnownTypes(object input, out object output)
             }
 
             var inputType = input.GetType().GetTypeInfo();
-            if (this.TrySerializeJavaScriptConverter(input, out output, inputType))

```

Comment:
That spits out a `TypeInfo` which is what I can't be using, I need the `Type`. We could split it up so we do something like

```
var foo = input.GetType();
var bar = foo.GetTypeInfo();
```

and use `foo` and `bar` when needed. Would safe us two calls to `.AsType()` ?